tssc/utils/containers.py

```diff
@@ -0,0 +1,243 @@
+                containers_config_auth_file=containers_config_auth_file
+            )
+
+def container_registry_login(
+    container_registry_uri,
+    container_registry_username,
+    container_registry_password,
+    container_registry_tls_verify=True,
+    containers_config_auth_file=None
+):
+    """Performs the login for a single container registry.
+
+    Requires one of the following to be installed to do the authentication:
+    * buidlah
+    * podman
+    * skopeo
+
+    Parametes
+    ---------
+    container_registry_uri : str or ConfigValue
+        URI to the container registry to log into.
+    container_registry_username : str or ConfigValue
+        Username to log into the container registry with.
+    container_registry_password : str or ConfigValue
+        Password to log into the container registry with.
+    container_registry_tls_verify : bool or str or ConfigValue
+        True to verify container registry certificates as part of authenticating.
+        False to ignore certificate chain.
+        NOTE: no matter what SSL is used to authenticate with container registry
+    containers_config_auth_file : str or ConfigValue, optional
+        Path of the authentication file.
+        If not specified default of the underlying authentication system will be used.
+
+    Raises
+    ------
+    RuntimeError
+        When can not find tool to login to container registry with.
+        When error loging into container registry.
+
+    See Also
+    --------
+    container_registries_login : authenticate with multiple container registries
+    sh.buildah : https://www.mankier.com/1/buildah-login
+    sh.podman : https://www.mankier.com/1/podman-login
+    sh.skopeo : https://www.mankier.com/1/skopeo-login
+    """
+
+    assert container_registry_uri
+    assert container_registry_username
+    assert container_registry_password
+
+    if isinstance(container_registry_uri, ConfigValue):
+        container_registry_uri = container_registry_uri.value
+    if isinstance(container_registry_username, ConfigValue):
+        container_registry_username = container_registry_username.value
+    if isinstance(container_registry_password, ConfigValue):
+        container_registry_password = container_registry_password.value
+    if isinstance(container_registry_tls_verify, ConfigValue):
+        container_registry_tls_verify = container_registry_tls_verify.value
+    if isinstance(containers_config_auth_file, ConfigValue):
+        containers_config_auth_file = containers_config_auth_file.value
+
+    # can use any of these tools to authenticate, look for them all and use first available
+    #
+    # NOTE: this all works because these three commands take the exact same parameters for login
+    # if implementing some new command, like docker, you will need to deal with the differences
+    buildah_path = sh.which('buildah')
+    podman_path = sh.which('podman')
+    skopeo_path = sh.which('skopeo')
+    if buildah_path is not None:
```

Comment:
couldn't you use a switch here?

Reply:
no such thing as switch in python. and even if there was....ummm....no, its not one input.